### PR TITLE
Wait for serving connection in entrypoint.sh

### PR DESCRIPTION
**Pull request by snormore**

### CODE REVIEW

**Comment by snormore** [comment]
Also added building of non-optimized Docker images to `script/distribute-docker`.

File: docker/gpu-optimized.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM triage/python2.7-tensorflow-serving-optimized-gpu:latest
+
+RUN pip install --upgrade pip && \
+    pip install tensorflow-serving-client ml-tools
+
```
Comment (stephensolis) [inline docker/gpu-optimized.Dockerfile]: +netcat here

File: tests/test_evaluation.py [hunk 2]
```diff
@@ -25,9 +25,21 @@ def test_eval_run_mobilenet_v1():
         assert len(image_predictions) == 2
 
         assert image_predictions[0]['image_path'] == 'cat.jpg'
-        assert image_predictions[0]['predictions'][:5] == [2.563341183758894e-07, 7.670701052120421e-06, 6.305415212182197e-08, 2.3645635849334212e-07, 2.3187270414837258e-07]
+        assert image_predictions[0]['predictions'][:5] == [
```
Comment (stephensolis) [inline tests/test_evaluation.py]: Would this need to use (eg) `numpy.testing.assert_array_almost_equal_nulp`?

File: tests/test_evaluation.py [hunk 3]
```diff
@@ -25,9 +26,27 @@ def test_eval_run_mobilenet_v1():
         assert len(image_predictions) == 2
 
         assert image_predictions[0]['image_path'] == 'cat.jpg'
-        assert image_predictions[0]['predictions'][:5] == [2.563341183758894e-07, 7.670701052120421e-06, 6.305415212182197e-08, 2.3645635849334212e-07, 2.3187270414837258e-07]
+        np.testing.assert_array_almost_equal_nulp(
+            np.array(image_predictions[0]['predictions'][:5]),
+            np.array([
+                2.5633391942392336e-07,
```
Comment (stephensolis) [inline tests/test_evaluation.py]: These might be too small for `assert_array_almost_equal_nulp` to be meaningful, can you test the top 5 predictions instead?

File: tests/test_evaluation.py [hunk 4]
```diff
@@ -25,9 +26,53 @@ def test_eval_run_mobilenet_v1():
...
-        assert image_predictions[0]['predictions'][:5] == [2.563341183758894e-07, 7.670701052120421e-06, 6.305415212182197e-08, 2.3645635849334212e-07, 2.3187270414837258e-07]
+        predictions = image_predictions[0]['predictions']
+        predictions = sorted(enumerate(predictions), key=lambda kv: kv[1], reverse=True)
+        predictions = [(imagenet_dictionary[i], score) for i, score in predictions][:5]
+        print(predictions)
+        classes = [name for name, _ in predictions]
+        assert classes == [
+            'tiger cat',
+            'Egyptian cat',
+            'tabby, tabby cat',
```
Comment (snormore) [inline tests/test_evaluation.py]: 🐱

File: tests/test_evaluation.py [hunk 5]
```diff
@@ -25,9 +26,53 @@ def test_eval_run_mobilenet_v1():
...
-        assert image_predictions[1]['predictions'][:5] == [5.521732191482442e-07, 1.7991060303756967e-05, 4.506294146722212e-07, 2.429260881342543e-08, 6.63468711081805e-08]
+        predictions = image_predictions[1]['predictions']
+        predictions = sorted(enumerate(predictions), key=lambda kv: kv[1], reverse=True)
+        predictions = [(imagenet_dictionary[i], score) for i, score in predictions][:5]
+        print(predictions)
+        classes = [name for name, _ in predictions]
+        assert classes == [
+            'golden retriever',
+            'Labrador retriever',
+            'beagle',
```
Comment (snormore) [inline tests/test_evaluation.py]: :dog:

**Pull request merged**